shipyard-controller/db/mongodb_event_repo.go

```diff
@@ -221,6 +224,96 @@ func (mdbrepo *MongoDBEventsRepo) DeleteEventCollections(project string) error {
 	return nil
 }
 
+func (e *MongoDBEventsRepo) GetStartedEventsForTriggeredID(eventScope *models.EventScope) ([]models.Event, error) {
+	startedEventType, err := keptnv2.ReplaceEventTypeKind(eventScope.EventType, string(common.StartedEvent))
+	if err != nil {
+		return nil, err
+	}
+	// get corresponding 'started' event for the incoming 'finished' event
+	filter := common.EventFilter{
+		Type:        startedEventType,
+		TriggeredID: &eventScope.TriggeredID,
+	}
+	return e.GetEventsWithRetry(eventScope.Project, filter, common.StartedEvent, maxRepoReadRetries)
+}
+
+func (e *MongoDBEventsRepo) GetEventsWithRetry(project string, filter common.EventFilter, status common.EventStatus, nrRetries int) ([]models.Event, error) {
+	log.Info(string("Trying to get " + status + " events"))
+	for i := 0; i <= nrRetries; i++ {
+		startedEvents, err := e.GetEvents(project, filter, status)
```

Comment:
let's rename this to `events` as the returned events here are not necessarily only `startedEvents`